cupy/core/core.pyx

```diff
@@ -1993,6 +1999,187 @@ cpdef ndarray dot(ndarray a, ndarray b, ndarray out=None):
     return tensordot_core(a, b, out, n, m, k, ret_shape)
 
 
+cpdef ndarray _get_all_addresses(size_t start_adr,
+                                 vector.vector[size_t] shape,
+                                 vector.vector[size_t] strides):
+    idx = numpy.array([start_adr])
```

Comment:
I thought, that this code must be faster on cpu, than on gpu.
If it has a similar speed I can change it. Or maybe someone know, how this should be written on gpu.